prow/cmd/pipeline/controller.go

```diff
@@ -690,6 +690,16 @@ func updateProwJobState(c reconciler, pj *prowjobv1.ProwJob, state prowjobv1.Pro
 			}
 			npj.Status.State = state
 			npj.Status.Description = msg
+
+			// Set the status build ID and URL if they're empty
+			for _, r := range runs {
+				if npj.Status.BuildID != "" && npj.Status.URL != "" {
```

Comment:
I would consider moving the if statement outside of `for`, it doesn't seem to be dependent on `r`.

Reply:
The reasoning here was to make sure we only do it once. =) But I'm fine with moving it.